### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by maxkudla**
- refactor cognitive_job_manager tests
- cover createCognitiveJob with tests
- cover reportProgress with tests

### CODE REVIEW

File: test/cognitive_job_manager.js [hunk 1]
```diff
@@ -28,131 +54,99 @@ contract('CognitiveJobManager', accounts => {
...
-        await workerInstance.alive({from: workerOwner});
-    });
-
-    it('should not create cognitive contract from outside of Pandora', async () => {
-
-        const numberOfBatches = 2;
-        const testDataset = await Dataset.new(datasetIpfsAddress, 1, numberOfBatches, 0, "m-a", "d-n");
-        const testKernel = await Kernel.new(kernelIpfsAddress, 1, 2, 3, "m-a", "d-n");
-
-        assertRevert(CognitiveJob.new(
```
Comment (Koroqe) [inline test/cognitive_job_manager.js]: Should we remove this test?

File: test/cognitive_job_manager.js [hunk 2]
```diff
@@ -81,43 +94,26 @@ contract('CognitiveJobManager', accounts => {
...
+        it(`should not create job if jobs count more than ${JOBS_COUNT_LIMIT}`, async () => {
 
+            const countForTest = 30;
+
+            let activeJobsCount = await pandora.cognitiveJobsCount();
 
             //lets create 30 jobs and finish them
-            for (let i = 0; i < 3; i++) {
+            for (let i = 1; i <= countForTest; i++) {
+                process.stdout.write(`      * should not create job if jobs count more than ${JOBS_COUNT_LIMIT}... ${i}/${countForTest}\r`);
```
Comment (Koroqe) [inline test/cognitive_job_manager.js]: Change description, please

File: test/cognitive_job_manager.js [hunk 3]
```diff
@@ -126,19 +122,156 @@ contract('CognitiveJobManager', accounts => {
...
+        it("should not hold payment from customer if job not putted into the queue", async () => {
+            const deposit_before = await pandora.deposits.call(customer);
+
+            await createCognitiveJob(pandora, 1, {from: customer});
+
+            await assersQueueDepth(pandora, 0);
+
+            const deposit_after = await pandora.deposits.call(customer);
+
+            assert.equal(deposit_before.toNumber(), deposit_after.toNumber(), 'Deposit was changed');
```
Comment (Koroqe) [inline test/cognitive_job_manager.js]: maybe "was NOT changed"? 
Also, you have to check customer's balance (will slightly decrease due gas costs)

File: test/cognitive_job_manager.js [hunk 4]
```diff
@@ -126,19 +122,156 @@ contract('CognitiveJobManager', accounts => {
...
+        it("should hold payment from customer if job was putted into the queue", async () => {
+            await createCognitiveJob(pandora, 1);
+
+            const deposit_before = await pandora.deposits.call(customer);
+
+            await createCognitiveJob(pandora, 2, {from: customer});
+
+            const deposit_after = await pandora.deposits.call(customer);
+
+            assert.notEqual(deposit_before.toNumber(), deposit_after.toNumber(), 'Deposit was not changed');
```
Comment (Koroqe) [inline test/cognitive_job_manager.js]: Add check account balance

File: test/cognitive_job_manager.js [hunk 5]
```diff
@@ -126,19 +122,156 @@ contract('CognitiveJobManager', accounts => {
...
+
+            await finishActiveJob(pandora, workerInstance0, workerOwner0);
+            await finishActiveJob(pandora, workerInstance1, workerOwner1);
+
+            await assersQueueDepth(pandora, 0);
```
Comment (Koroqe) [inline test/cognitive_job_manager.js]: Name typo

File: test/cognitive_job_manager.js (hunk 3 above)
Comment (maxkudla) [reply test/cognitive_job_manager.js]: this case test only deposit changes

File: test/cognitive_job_manager.js (hunk 4 above)
Comment (maxkudla) [reply test/cognitive_job_manager.js]: this case test only deposit changes

**Pull request merged**